algraph.py

```diff
@@ -0,0 +1,54 @@
+
+class ALGraph:
+    """adjacency_list graph"""
+
+    def __init__(self):
+        self.adjacency_list = {}
+        
+    def addEdges(self, edges):
+        for edge in edges:
+            self.addVerts(edge)
+            v1, v2 = edge
+            self.adjacency_list[v1].add(v2)
+            
+    def addVerts(self, verts):
+        for vert in verts:
+            if self.adjacency_list.get(vert) is None:
+                self.adjacency_list[vert] = set()
+    
+    def findTopologicalOrdering(self):
+        indegree = {}
+        for vert in self.adjacency_list.keys():
+            indegree[vert] = 0
```

Comment:
Purt sure that

`indegree = { vert:0 for vert in self.adjacency_list.keys() }`

is more pythonic